tests/async/test_page.py

```diff
@@ -1294,3 +1294,33 @@ def binding(source, o):
     await page.expose_binding("log", lambda source, o: binding(source, o))
     await page.evaluate("const a = {}; a.b = a; window.log(a)")
     assert binding_values[0]["b"] == binding_values[0]
+
+
+async def test_page_pause_should_reset_default_timeouts(
+    page: Page, headless: bool, server: Server
+) -> None:
+    if not headless:
+        pytest.skip()
+
+    await page.goto(server.EMPTY_PAGE)
+    page.pause()
+    with pytest.raises(Error, match="Timeout 30000ms exceeded."):
```

Comment:
Do we really want to slow down all tests by 30s?

Reply:
I think no other option instead of hacking around the impl for testing.